starter_code/queries.md

```diff
@@ -6,46 +6,76 @@
 
 <!-- Your Code Goes Here -->
 
+FILTER {name: "Babelgum"}
+PROJECT {name: 1}
+
 ### 2. All the companies that have more than 5000 employees. Limit the search to 20 companies and sort them by **number of employees**.
 
 <!-- Your Code Goes Here -->
-
+FILTER { number_of_employees: { $gt: 5000 } }
+SORT { number_of_employees: 1 }
+LIMIT 20
 ### 3. All the companies founded between 2000 and 2005, both years included. Retrieve only the `name` and `founded_year` fields.
 
 <!-- Your Code Goes Here -->
 
+FILTER {$and: [{founded_year: {$gte: 2000}}, {founded_year: {$lte: 2005}}]}
+
 ### 4. All the companies that had a Valuation Amount of more than 100.000.000 and have been founded before 2010. Retrieve only the `name` and `ipo` fields.
 
 <!-- Your Code Goes Here -->
 
+FILTER {$and: [{"ipo.valuation_amount": {$gt: 100000000}}, {founded_year: {$lt: 2010}}]}
+PROJECT {name: 1, ipo: 1, _id:0}
+
 ### 5. All the companies that have less than 1000 employees and have been founded before 2005. Order them by the number of employees and limit the search to 10 companies.
 
 <!-- Your Code Goes Here -->
 
+FILTER {$and: [{number_of_employees: {$lt: 1000}}, {founded_year: {$lt: 2005}}]}
+SORT {number_of_employees: 1}
+
+LIMIT 10
+
 ### 6. All the companies that don't include the `partners` field.
 
 <!-- Your Code Goes Here -->
 
+FILTER {partners: { $exists:false}}
+
 ### 7. All the companies that have a null type of value on the `category_code` field.
 
 <!-- Your Code Goes Here -->
 
+FILTER {category_code:{ $type:"null"}}
+
 ### 8. All the companies that have at least 100 employees but less than 1000. Retrieve only the `name` and `number of employees` fields.
 
 <!-- Your Code Goes Here -->
 
+FILTER  {$and: [{number_of_employees: {$gte: 100}}, {number_of_employees: {$lt: 1000}}]}
+PROJECT {name: 1, number_of_employees: 1}
+
+
 ### 9. Order all the companies by their IPO price descendently.
 
 <!-- Your Code Goes Here -->
 
+SORT {"ipo.valuation_amount": -1}
+
 ### 10. Retrieve the 10 companies with more employees, order by the `number of employees`
 
 <!-- Your Code Goes Here -->
+SORT {number_of_employees:-1}
+LIMIT 10
 
 ### 11. All the companies founded on the second semester of the year. Limit your search to 1000 companies.
 
 <!-- Your Code Goes Here -->
 
+FILTER {founded_month: {$gt: 6}}
```

Comment:
Pour être sûr du résultat, on peut aussi ajouter la limite haute de la manière suivante:
FILTER {$and: [{founded_month: {$gte: 6}}, {founded_month: {$lte: 12}}] }